tests/helpers/test_event.py

```diff
@@ -4048,63 +4050,82 @@ async def test_periodic_task_leaving_dst_2(hass, freezer):
-    assert p_action is action
-    assert p_point == now + timedelta(seconds=3)
+    assert await asyncio.wait_for(future, delay + delay_tolerance)
 
 
 async def test_async_call_later(hass):
     """Test calling an action later."""
 
-    def action():
-        pass
+    future = asyncio.get_running_loop().create_future()
+    scheduletime = dt_util.now()
+    delay = 0.5
+    delay_tolerance = 0.1
 
-    now = datetime(2017, 12, 19, 15, 40, 0, tzinfo=dt_util.UTC)
+    def action(__now: datetime):
+        result = __now > scheduletime + timedelta(seconds=delay)
+        result &= __now < scheduletime + timedelta(seconds=delay + delay_tolerance)
+        future.set_result(result)
 
-    with patch(
-        "homeassistant.helpers.event.async_track_point_in_utc_time"
-    ) as mock, patch("homeassistant.util.dt.utcnow", return_value=now):
-        remove = async_call_later(hass, 3, action)
+    remove = async_call_later(hass, delay, action)
 
-    assert len(mock.mock_calls) == 1
-    p_hass, p_action, p_point = mock.mock_calls[0][1]
-    assert p_hass is hass
-    assert p_action is action
-    assert p_point == now + timedelta(seconds=3)
-    assert remove is mock()
+    assert await asyncio.wait_for(future, delay + delay_tolerance)
+    assert isinstance(remove, typing.Callable)
+    remove()
 
 
 async def test_async_call_later_timedelta(hass):
     """Test calling an action later with a timedelta."""
 
-    def action():
-        pass
+    future = asyncio.get_running_loop().create_future()
+    scheduletime = dt_util.now()
+    delay = 0.35
+    delay_tolerance = 0.1
 
-    now = datetime(2017, 12, 19, 15, 40, 0, tzinfo=dt_util.UTC)
+    def action(__now: datetime):
+        result = __now > scheduletime + timedelta(seconds=delay)
+        result &= __now < scheduletime + timedelta(seconds=delay + delay_tolerance)
+        future.set_result(result)
 
-    with patch(
-        "homeassistant.helpers.event.async_track_point_in_utc_time"
-    ) as mock, patch("homeassistant.util.dt.utcnow", return_value=now):
-        remove = async_call_later(hass, timedelta(seconds=3), action)
-
-    assert len(mock.mock_calls) == 1
-    p_hass, p_action, p_point = mock.mock_calls[0][1]
-    assert p_hass is hass
-    assert p_action is action
-    assert p_point == now + timedelta(seconds=3)
-    assert remove is mock()
+    remove = async_call_later(hass, timedelta(seconds=delay), action)
+
+    assert await asyncio.wait_for(future, delay + delay_tolerance)
+    assert isinstance(remove, typing.Callable)
+    remove()
+
+
+async def test_async_call_later_cancel(hass):
+    """Test canceling a call_later action."""
+
+    future = asyncio.get_running_loop().create_future()
+    dt_util.now()
+    delay = 0.25
+    delay_tolerance = 0.1
+
+    def action(__now: datetime):
+        future.set_result(False)
+
+    remove = async_call_later(hass, timedelta(seconds=delay), action)
+    # wait just a bit..
+    await asyncio.sleep(0.05)
```

Comment:
Can you avoid sleeping and use `async_fire_time_changed` instead in these tests?

Reply:
Hello,
It took a bit to grasp the `async_fire_time_changed` but that's definitely awesome to use (hoping I got it right!)
tests are now still awaiting for timeout when checking the outcome but it is just a 'safety' measure since the `async_fire_time_changed` has already rolled out the time.
I've also updated the api code to benefit from the tests framework

Reply:
P.S.
sorry to bother you, I should likely start following the Discord chat but, is there any way to speed up the 'pre-commit' ?
my vscode runs indefinitely before being able to actually commit these tiny modifications (like 10 minutes or so..my machine is really 'aged' even being a gaming laptop)

Reply:
Thank you for the quick suggestion.
I see some tests over other parts of HA are failing and I'm trying to figure out a reasonable 'patch' to the relative tests since I guess the issue lies in the mock over the timing functions for those failing...I'm sure I still can't grab all of the nuances of the framework so I'm really trying to study and understand